### ISSUE TITLE
Wrong alias in WHERE clause if using Fetch and scalar Select

### ISSUE THREAD

**Issue by filronius** [nhibernate/nhibernate-core#3263]
I have two entities (Employee and OptionaInfo) which are linked over HasOne mappings. When running this query:
```C#
 _session.Query<Employee>()
    .Fetch(x => x.OptionalInfo)
    .Where(x => x.OptionalInfo != null)
    .Select(x => x.OptionalInfo.Age)
    .ToList();
```
I get this error:
```
could not execute query
[
    select optionalin1_.Age as col_0_0_ 
    from Employee employee0_
    left outer join OptionalInfo optionalin1_ on employee0_.EmployeeId=optionalin1_.EmployeeId
    where optionalin2_.EmployeeId is not null
]
PostgresException: 42P01: missing FROM-clause entry for table »optionalin2_«
```
NHibernate uses alias `optionalin2_` but doesn’t define it. If I change `optionalin2_` to `optionalin1_` manually, the query works.
The error only occurs if the Fetch method is present. Without Fetch everything works as expected.
I tested NHibernate versions 5.4.1, 5.4.0 and 5.3.15. The error appears in 5.4.1 and 5.4.0. Version 5.3.15 works just fine. I'm using PostgreSQL.

Here are my entities and maps:
```C#
public class Employee
{
    public virtual int EmployeeId { get; set; }
    public virtual string Name { get; set; }
    public virtual OptionalInfo? OptionalInfo { get; set; }
}

public class OptionalInfo
{
    public virtual int EmployeeId { get; set; }
    public virtual int Age { get; set; }
    public virtual Employee Employee { get; set; }
}

public class EmployeeMap : ClassMap<Employee>
{
    public EmployeeMap()
    {
       Table("Employee");
        Id(x => x.EmployeeId).GeneratedBy.Identity();
        Map(x => x.Name).Not.Nullable();
        HasOne(x => x.OptionalInfo).Cascade.SaveUpdate();
    }
}

public class OptionalInfoMap : ClassMap<OptionalInfo>
{
    public OptionalInfoMap()
    {
        Table("OptionalInfo");
        Id(x => x.EmployeeId).GeneratedBy.Foreign("Employee");
        Map(x => x.Age).Not.Nullable();
        HasOne(x => x.Employee).Cascade.None();
    }
}
```

**Comment by bahusoid** [comment]
> The error only occurs if the Fetch method is present

Might be regression from https://github.com/nhibernate/nhibernate-core/pull/2496

**Comment by bahusoid** [comment]
But why Fetch is present in query if your Select some scalar values? While it's a valid issue your Fetch makes no sense in this query.

**Comment by filronius** [comment]
Yes, the Fetch makes no sense in this case.

The project, in which I noticed this bug, is more complex than the example above and has included Fetch already. Unfortunately I can’t remove it there.

**Comment by bahusoid** [comment]
Fixed by #3267

**Comment by filronius** [comment]
Thank you!
